OSC fails to build with --enable-heterogeneous

With gcc on master tip:

```
  CC       osc_pt2pt_active_target.lo
In file included from osc_pt2pt.h:43:0,
                 from osc_pt2pt_active_target.c:26:
osc_pt2pt_active_target.c: In function 'ompi_osc_pt2pt_complete':
osc_pt2pt_active_target.c:353:39: error: 'proc' undeclared (first use in this function)
         osc_pt2pt_hton(&complete_req, proc);
                                       ^
osc_pt2pt_header.h:361:58: note: in definition of macro 'osc_pt2pt_hton'
     osc_pt2pt_hton_intr((ompi_osc_pt2pt_header_t *)(h), (p));
                                                          ^
osc_pt2pt_active_target.c:353:39: note: each undeclared identifier is reported only once for each function it appears in
         osc_pt2pt_hton(&complete_req, proc);
                                       ^
osc_pt2pt_header.h:361:58: note: in definition of macro 'osc_pt2pt_hton'
     osc_pt2pt_hton_intr((ompi_osc_pt2pt_header_t *)(h), (p)); 
                                                          ^
make[2]: *** [osc_pt2pt_active_target.lo] Error 1
```

@hjelmn Can you have a look?

--- a/ompi/mca/osc/pt2pt/osc_pt2pt_active_target.c
+++ b/ompi/mca/osc/pt2pt/osc_pt2pt_active_target.c
@@ -336,7 +336,8 @@ int ompi_osc_pt2pt_complete (ompi_win_t *win)
     for (size_t i = 0 ; i < group_size ; ++i) {
         ompi_osc_pt2pt_header_complete_t complete_req;
         int rank = peers[i]->rank;
-
+        ompi_proc_t *proc = ompi_comm_peer_lookup (module->comm, rank);
+        
         if (my_rank == rank) {
             /* shortcut for self */
             osc_pt2pt_incoming_complete (module, rank, 0);

